java/src/main/native/OrtJniUtil.c

```diff
@@ -518,22 +501,27 @@ size_t copyJavaToTensor(JNIEnv *jniEnv, ONNXTensorElementDataType onnxType, uint
         // recurse through the dimensions
         // Java arrays are objects until the final dimension
         jobjectArray inputObjArr = (jobjectArray) input;
-        uint32_t dimLength = (*jniEnv)->GetArrayLength(jniEnv,inputObjArr);
-        size_t sizeConsumed = 0;
-        for (uint32_t i = 0; i < dimLength; i++) {
-            jarray childArr = (jarray) (*jniEnv)->GetObjectArrayElement(jniEnv,inputObjArr,i);
-            sizeConsumed += copyJavaToTensor(jniEnv, onnxType, tensor + sizeConsumed, tensorSize - sizeConsumed, dimensionsRemaining - 1, childArr);
+        int32_t dimLength = (*jniEnv)->GetArrayLength(jniEnv, inputObjArr);
+        int64_t sizeConsumed = 0;
+        for (int32_t i = 0; i < dimLength; i++) {
+            jarray childArr = (jarray) (*jniEnv)->GetObjectArrayElement(jniEnv, inputObjArr, i);
+            int64_t consumed = copyJavaToTensor(jniEnv, onnxType, tensor + sizeConsumed, tensorSize - sizeConsumed, dimensionsRemaining - 1, childArr);
+            sizeConsumed += consumed;
             // Cleanup reference to childArr so it doesn't prevent GC.
-            (*jniEnv)->DeleteLocalRef(jniEnv,childArr);
+            (*jniEnv)->DeleteLocalRef(jniEnv, childArr);
+            // If we failed to copy an array then break and return.
+            if (consumed == -1) {
+              return -1;
+            }
```

Comment:
Should we throw here?

Reply:
`copyJavaToPrimitiveArray` has already thrown if we get -1 here, so we're just unwinding the stack.